### PR TITLE
EFA-618 Implementation of new passphrase handling

### PR DESCRIPTION
**Pull request by ig-garcia**

### CODE REVIEW

File: src/cxx/basic_api.cc [hunk 1]
```diff
@@ -468,6 +468,32 @@ JNIEXPORT void JNICALL Java_foundation_pEp_jniadapter_Engine__1config_1passphras
...
+
+    bool _enable = static_cast<bool>(enable);
+    const char *_passphrase = to_string(env, passphrase);
+    const char *_email = to_string(env, email);
+
+    if (_enable) {
+        passphrase_cache.add(_email, _passphrase);
+    }
+    else {
+        passphrase_cache.remove(_passphrase);
```
Comment (ghost) [inline src/cxx/basic_api.cc]: That looks wrong to me. The passphrase_cache method here expects the account email, you provide the passphrase.

File: src/cxx/basic_api.cc [hunk 2]
```diff
@@ -468,6 +468,32 @@ JNIEXPORT void JNICALL Java_foundation_pEp_jniadapter_Engine__1config_1passphras
     }
 }
 
+JNIEXPORT void JNICALL Java_foundation_pEp_jniadapter_Engine__1config_1passphrase_1for_1new_1keys_1by_1email(JNIEnv *env,
+                                                                                                  jobject obj,
+                                                                                                  jboolean enable,
+                                                                                                  jbyteArray email,
+                                                                                                  jbyteArray passphrase)
```
Comment (ghost) [inline src/cxx/basic_api.cc]: That's one weird style of indentation (just saying, this is just cosmetics).

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/cxx/basic_api.cc (hunk 1 above)
Comment (ig-garcia) [reply src/cxx/basic_api.cc]: Thanks for that, updated.

File: src/cxx/basic_api.cc (hunk 2 above)
Comment (ig-garcia) [reply src/cxx/basic_api.cc]: Updated to match the style of the file more or less.

**Review by ghost — APPROVED** [review]

**Pull request merged**